test/unit/math/prim/mat/prob/vector_rng_test_helper.hpp

```diff
@@ -0,0 +1,364 @@
+#include <stan/math/prim/mat.hpp>
+#include <gtest/gtest.h>
+#include <boost/math/distributions.hpp>
+#include <boost/random/mersenne_twister.hpp>
+#include <test/unit/math/prim/scal/prob/util.hpp>
+
+enum class Constraint { None, Positive, NonNegative };
+
+/*
+ * Resize v to be length N and fill v with values that satisfy the given constraint c.
+ *
+ * @tparam T1 Type of v
+ * @param v Vector to fill with values
+ * @param c Constraint that values must fulfill
+ * @param N Number of elements that v will be resized to
+ */
+template <typename T1>
+void generateValues(T1& v, Constraint c, int N) {
+  v.resize(N);
+  
+  if (c == Constraint::None) {
+    for (int i = 0; i < N; i++)
+      v[i] = i - 1;
+  } else if (c == Constraint::Positive) {
+    for (int i = 0; i < N; i++)
+      v[i] = i + 1;
+  } else if (c == Constraint::NonNegative) {
+    for (int i = 0; i < N; i++)
+      v[i] = i;
+  }
+}
+
+/*
+ * Set v to a value that satisfies the constraint (and as much as possible
+ * does not satisfy other constraints -- but this isn't totally possible)
+ *
+ * @tparam T1 Type of v
+ * @param v Variable to assign value to
+ * @param c Constraint that value must fulfill
+ * @param N Unused
+ */
+template <>
+void generateValues(double& v, Constraint c, int N) {
+  if (c == Constraint::None) {
```

Comment:
Also, I just took a look at some code of Bob's that does similar stuff for some other work and it might be interesting to see how he approached a similar goal: https://github.com/stan-dev/math/blob/develop/test/unit/math/mix/mat/util/autodiff_tester.hpp#L354